File: 01.js
const util = require('./util.js')

const targetValue = 2020
const today = 1

async  function main() {
    var input = await util.getInput(today);
    const lines = input
        .split('\n')
        .map(Number)
            
    var answer1 = part1(lines)
    console.log("Part 1: " + answer1)
    
    var answer2 = part2(lines)
    console.log("Part 2: " + answer2)

    // sendAnswers(answer1, answer2)
}


function part1(lines) {
    for(var i = 0; i < lines.length - 1; i++) {
        for(var j = i+1; j < lines.length; j++) {
            if(targetValue === lines[i] + lines[j]) {
                return lines[i] * lines[j]
            }
        }
    }
}

function part2(lines) {
    for(var i=0; i<lines.length - 2; i++) {
        for(var j = i + 1; j < lines.length - 1; j++) {
            for(var k = j + 1; k < lines.length; k++) {
                if(targetValue === lines[i] + lines[j] + lines[k]) {
                    return lines[i] * lines[j] * lines[k]
                }
            }
        }
    }
}

function sendAnswers(answer1, answer2) {
    util.sendAnswer(today, 1, answer1)
        .then(console.log)
        .catch(console.error)

    util.sendAnswer(today, 2, answer2)
        .then(console.log)
        .catch(console.error)
}


main();

File: 01.go
package main

import (
	"fmt"
	"io/ioutil"
	"strings"
)

const (
	today  = 1
	target = 2020
)

func main() {
	// Read todays file
	dat, err := ioutil.ReadFile(fmt.Sprintf("%02d.in", today))
	if err != nil {
		panic(err)
	}

	// Split lines, and parse as int
	var intLines = mapToInt(strings.Split(string(dat), "\n"))

	for i, line := range intLines {
		for j := i + 1; j < len(intLines); j++ {

			// Search for Part 1 duplet
			if line+intLines[j] == target {
				fmt.Printf("Part1 %d\n", line*intLines[j])
			}

			// Search for Part 2 triplet
			for k := j + 1; k < len(intLines); k++ {
				if line+intLines[j]+intLines[k] == target {
					fmt.Printf("Part2 %d\n", line*intLines[j]*intLines[k])
				}
			}
		}
	}
}
